### ISSUE TITLE
Prompt is slow after multiple "source ~/.zshrc"

### ISSUE THREAD

**Issue by silverneko** [zsh-users/zsh-autosuggestions#470]
### Describe the bug
zsh prompt becomes very slow after multiple times of "source ~/.zshrc". I suspect it's the widgets of "_zsh_autosuggest_bound_#n_*" slowing zsh down, because after every time I do "source ~/.zshrc" the number of these widgets increases. For example when a new shell starts only widgets of "_zsh_autosuggest_bound_1_*" are registered, and after a second "source ~/.zshrc" widgets of "_zsh_autosuggest_bound_2_*" are added.

I use oh-my-zsh to load this plugin.

### To Reproduce
Do this for 10 to 20 times:
```
$ source ~/.zshrc
```

### Desktop
 - OS + distribution: debian
 - Zsh version: 5.7
 - Plugin version: 43f3bc40

**Comment by ericfreese** [comment]
I could see this being an issue but will probably be low priority for me to fix since the issue is avoidable by not re-sourcing zshrc many times.

**Comment by silverneko** [comment]
You're right! It's not really a issue for normal users. I only encounter this issue because I'm testing with different .zshrc configurations thus sourcing .zshrc multiple times.

**Comment by z0rc** [comment]
Sourcing `.zshrc` isn't optimal in many ways and will break more than z-asug. Instead it's better to use `exec zsh` to completely reinit from clean state.

**Comment by NightMachinary** [comment]
@silverneko How do you check the list of all widgets?

**Comment by silverneko** [comment]
`zle -l` or if you have auto completion then `_<tab>`
